docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,861 @@
+# Installing OpenShift on OpenStack User-Provisioned Infrastructure
+
+The User-Provisioned Infrastructure (UPI) process installs OpenShift in stages, providing opportunities for modifications or integrating with existing infrastructure.
+
+It contrasts with the fully-automated Installer-Provisioned Infrastructure (IPI) which creates everything in one go.
+
+With UPI creating the cloud (OpenStack) resources (e.g. Nova servers, Neutron ports, security groups) is the responsibility of the person deploying OpenShift.
+
+The installer is still used to generate the ignition files and monitor the installation process.
+
+This provides a greater flexibility at the cost of a more explicit and interactive process.
+
+
+## Prerequisites
+
+The requirements for UPI are broadly similar to the [ones for OpenStack IPI][ipi-reqs]:
+
+[ipi-reqs]: ./openstack#openstack-requirements
+
+- OpenStack account with `clouds.yaml`
+  - This document will use a cloud called `openstack`
+- Nova flavors
+  - This document will use `m1.xlarge` for masters and `m1.large` for workers
+- An external subnet you want to use for floating IP addresses
+  - This document will use `external`
+- The `openshift-install` binary in your `$PATH`
```

Comment:
or `%PATH%`?

Reply:
That's Windows syntax, isn't it? The installer binaries are only available for linux and macos for now. I'm sure there are custom shells that use a different syntax, but `$ENV_VAR_NAME` is the posix-like thing everyone reading this should understand.

Reply:
> `$ENV_VAR_NAME` is the posix-like thing everyone reading this should understand.

I agree 100%. I just thought that a wink to Powershell (or whatever it's called these days) would hint our Windows-based friends that this doc works for them as well. But apparently it doesn't! I wrongly assumed we were providing the Windows binary.

Reply:
_please consider as resolved_

Reply:
Gotcha. I wasn't sure myself -- had to check the release page.